### PR TITLE
chore: Add development / offline mode compile time flag

### PR DESCRIPTION
**Pull request by hxr404**
This PR adds a compile time flag which allows to disable the (now) mandatory Microsoft login.
It would make developing and testing PrismLauncher much easier.
I personally find it quite a hassle having to log in to my Microsoft account every single time I recompile (and deleting application state in that process to have a clean testing environment).
Commenting out the function just to be quicker at testing stuff seems unnecessarily inconvenient and increases the risk of merge conflicts.
I think issues like this one really are a perfect match for using compile time flags.

I understand that this may be met with some controversy, however I'd like to mention that this does NOT enable piracy per-se.
The flag is disabled by default, so this can only be used if people compile from source.
Users wanting to pirate can do this already by simply removing / commenting out the check and compiling from source.
So from their perspective it doesn't change anything, this change only benefits people who want to use it for legitimate reasons.

### CODE REVIEW

**Comment by theofficialgman** [comment]
if you are truely a dev then you should already know know how to copy your login credentials between launchers

close this just as this was closed https://github.com/PrismLauncher/PrismLauncher/pull/974

**Comment by hxr404** [comment]
As stated in the PR's description, having to work with application state is inconvenient, especially when working on breaking changes (which make application state (like login creds) not being portable/copy&paste-able)

Btw: how does this PR relate to anything proposed in #974?
Removing the Account check would at best be a really bad, if not horrible idea, so obviously I do not support such changes. Again, this PR does NOT enable piracy, in the worst case it could make it slightly easier by saving a malicious user a few keystrokes.

Imo there is no real reason why such a flag shouldn't be added. And my usecase definetly isn't the only one.

There are many possible advantages to including this flag, it would enable the PrismLauncher devs to have more control over how their software is used:
For instance it would be possible to remove branding, links and any other mentions of prismlauncher automatically if the flag is enabled.
And that would only improve the current situation with dozens of "cracked" versions which don't even bother to remove prismlauncher's api keys.
Bad guys who want to pirate this will always find a way to do it and we have no way of preventing it, so the best we can do is trying to reduce the damage done.

**TL;DR:** this PR doesn't change anything about the fact that (unmodified) prismlauncher is paid-only. It enables the possibility to reduce harm done by piracy and has other legitimate uses too. There are no negative causes.

Apr 16, 2023 01:14:36 theofficialgman ***@***.***>:

**Comment by DioEgizio** [comment]
There is literally 0 need for that except piracy
